services/system/Accounts/ui/src/AccountSelection.tsx

```diff
@@ -340,82 +212,44 @@ export const AccountSelection = () => {
-          </div>
-          <CardTitle className="text-center">Loading...</CardTitle>
-        </CardHeader>
-      </Card>
-    );
+    return <LoadingCard />;
   }
 
   if (inviteToken?.state == "accepted" && !isInviteClaimed) {
-    return (
-      <Card className="w-[350px] mx-auto mt-4">
-        <CardHeader>
-          <div className="mx-auto">
-            <TriangleAlert className="w-12 h-12" />
-          </div>
-          <CardTitle>Invitation already accepted.</CardTitle>
-          <CardDescription>
-            This invitation has been accepted by{" "}
-            <span className="text-primary font-semibold">
-              {inviteToken.actor}
-            </span>
-            .
-          </CardDescription>
-        </CardHeader>
-      </Card>
-    );
+    return <InviteAlreadyAcceptedCard token={token} />;
   }
 
   if (inviteToken?.state == "rejected") {
-    return (
-      <Card className="w-[350px] mx-auto mt-4">
-        <CardHeader>
-          <div className="mx-auto">
-            <TriangleAlert className="w-12 h-12" />
-          </div>
-          <CardTitle>Invitation rejected.</CardTitle>
-          <CardDescription>
-            This invitation has been rejected
-            {inviteToken?.actor && inviteToken.actor !== "invite-sys"
-              ? ` by ${inviteToken.actor}.`
-              : "."}
-          </CardDescription>
-        </CardHeader>
-      </Card>
-    );
+    return <InviteRejectedCard token={token} />;
   }
 
-  const expiryMessage =
-    inviteToken &&
-    `This invitation expired ${dayjs().to(inviteToken.expiry)} (${dayjs(
-      inviteToken.expiry
-    ).format("YYYY/MM/DD HH:mm")}).`;
-
   if (isExpired) {
-    return (
-      <Card className="w-[350px] mx-auto mt-4">
-        <CardHeader>
-          <div className="mx-auto">
-            <AlarmClockMinus className="w-12 h-12" />
-          </div>
-          <CardTitle>Expired invitation</CardTitle>
-          <CardDescription>{expiryMessage}</CardDescription>
-          <CardDescription>
-            Please ask the sender{" "}
-            <span className="text-primary">{inviteToken?.inviter}</span> for a
-            new one.
-          </CardDescription>
-        </CardHeader>
-      </Card>
-    );
+    return <InviteExpiredCard token={token} />;
   }
 
+  const onAcceptOrLogin = () => {
+    if (isInvite) {
+      if (!inviteToken) {
+        throw new Error(`Expected invite token loaded`);
+      }
+      acceptInvite({
+        token: z.string().parse(token),
+        accountName: z.string().parse(selectedAccount?.account),
+        app: inviteToken.app,
+        origin: inviteToken.appDomain,
+      });
+    } else {
+      if (!connectionToken) {
+        throw new Error(`Expected connection token for a login`);
+      }
+      login({
+        app: connectionToken.app,
+        origin: connectionToken.origin,
+        accountName: selectedAccount!.account,
+      });
+    }
+  };
```

Comment:
What happens to the UI if this function throws?

Reply:
Whatever happened before this refactor. The code is functionally the same; just rearranged.
But good question.